src/services/impl/GameContext.cpp

```diff
@@ -0,0 +1,29 @@
+#include "GameContext.hh"
+
+#include "RA_Log.h"
+#include "RA_StringUtils.h"
+
+#include "services\impl\FileTextReader.hh"
+#include "services\impl\FileTextWriter.hh"
+
+namespace ra {
+namespace services {
+namespace impl {
+
+void GameContext::LoadGame(unsigned int nGameId, const std::wstring& sGameTitle)
+{
+    if (nGameId == 0)
```

Comment:
It's up to you, but this what I think about this. An exception handling strategy can come later.

I think an ID of 0 should be considered undefined. If you used this class with iterators and the ID didn't exist, it'll throw an exception. Is it ever allowed for GameID to be 0? If not, you should probably throw an exception and add a new entry when it's caught as it's an exceptional circumstance that only occurs at runtime.

Yes, above I did say this looks like a constructor. Parameterized default constructors can throw, a parameterless/compiler-generated one should not.

Reply:
This is to handle the case where a hash doesn't resolve, or the user unloads the game without closing the emulator. It might make more sense if I had moved the server API call into this class, but I wanted to save that for a separate PR.